test/signature-unit-tests.spec.ts

```diff
@@ -91,9 +90,17 @@ describe("Signature unit tests", function () {
         "sH1gxKve8wlU8LlFVa2l6w3HMJ0=",
       ];
 
+      const firstGrandchild = doc.firstChild?.firstChild;
       // @ts-expect-error FIXME
-      for (let i = 0; i < sig.references.length; i++) {
+      if (xpath.isElement(firstGrandchild)) {
+        const matchedReference = sig.validateElementAgainstReferences(firstGrandchild, doc);
+        expect(matchedReference).to.not.be.false;
+      } else {
         // @ts-expect-error FIXME
+        expect(xpath.isElement(firstGrandchild)).to.be.true;
+      }
```

Comment:
Perhaps throwing an Error explicitly in the else block? Thanks for the explanation.

Reply:
I've added some assertion checks over [here](https://github.com/goto100/xpath/pull/128). I'll use those in the next go-around. Does that seem like it would work?

Reply:
I actually was referring to replacing line 100 of the test (ie, the assert that will always fail), but I think what you've done on xpath will work!